### ISSUE TITLE
Taskd hanging

### ISSUE THREAD

**Issue by djmitche** [GothenburgBitFactory/taskwarrior#2987]
The taskd service in freecinc is hanging after only tens of requests are served (sometimes 10 or so, sometimes up to 50).

I have long had a "liveness probe" that checks the TCP port for accessibility, and restarts the container if it fails -- but it takes a while for that to time out, leaving taskd nonresponsive in the interim.

Any advice for debugging this?  It's in a container, so I can't use `strace` to see what the process is up to (no CAP_SYS_PTRACE).

**Comment by dgrabla** [comment]
I observed this as well. It is a problem I had on my taskd for years. The server is there and pick up the requests, but it does not answer and the clients block till the connection timeouts. The solution that works for me is a periodic restart. I could not trace the issue.

**Comment by djmitche** [comment]
Thanks!  I've started running it on my VPS and I'm now having the same issues.  Maybe `taskdctl` in a crontask will help?

**Comment by dgrabla** [comment]
While taskdctl is without a doubt better than my horrible 'killall taskd; nohup taskd', it is still a nasty workaround. There must be a reason why the beast hangs.

**Comment by djmitche** [comment]
Taskd won't be continued.

**Comment by dgrabla** [comment]
@djmitche Sorry I've not been following the project. Is the recommended way to sync tasks to sync the files themselves? Should be using git to take over the merges?  Is taskd 'complete' as as such it will not need maintenance?

**Comment by djmitche** [comment]
It's been unmaintained for quite a while, and there's a replacement in the works, so it's unlikely this bug will ever be addressed. For the moment, taskd is still the recommended way, with occasional restarts due to hangs and memory leaks.
